File: src/engine/transpile/src/test/resources/golden/SYK001/ORD1_受注レコード.py
"""SYK001 のデータ部レコード ORD1-受注レコード を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class ORD1_受注レコード:
    """01 ORD1-受注レコード 総バイト長 253。"""

    LENGTH = 253

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(253)
        else:
            self._data = bytearray(data)
            if len(self._data) < 253:
                self._data.extend(b"\x00" * (253 - len(self._data)))

    @property
    def data(self):
        return self._data

    # ORD1-受注番号
    def get_ORD1_受注番号(self):
        return decode_alphanumeric(self._data, 0, 10)
    def set_ORD1_受注番号(self, value):
        encode_alphanumeric(self._data, 0, 10, value)

    # ORD1-受注日
    def get_ORD1_受注日(self):
        return decode_zoned(self._data, 10, 8)
    def set_ORD1_受注日(self, value):
        encode_zoned(self._data, 10, 8, False, value)

    # ORD1-受注日-YMD REDEFINES ORD1-受注日 (同一オフセット 10)

    # ORD1-受注日-年
    def get_ORD1_受注日_年(self):
        return decode_zoned(self._data, 10, 4)
    def set_ORD1_受注日_年(self, value):
        encode_zoned(self._data, 10, 4, False, value)

    # ORD1-受注日-月
    def get_ORD1_受注日_月(self):
        return decode_zoned(self._data, 14, 2)
    def set_ORD1_受注日_月(self, value):
        encode_zoned(self._data, 14, 2, False, value)

    # ORD1-受注日-日
    def get_ORD1_受注日_日(self):
        return decode_zoned(self._data, 16, 2)
    def set_ORD1_受注日_日(self, value):
        encode_zoned(self._data, 16, 2, False, value)

    # ORD1-得意先コード
    def get_ORD1_得意先コード(self):
        return decode_alphanumeric(self._data, 18, 6)
    def set_ORD1_得意先コード(self, value):
        encode_alphanumeric(self._data, 18, 6, value)

    # ORD1-受注金額合計
    def get_ORD1_受注金額合計(self):
        return decode_packed(self._data, 24, 6)
    def set_ORD1_受注金額合計(self, value):
        encode_packed(self._data, 24, 6, True, value)

    # ORD1-明細件数
    def get_ORD1_明細件数(self):
        return decode_packed(self._data, 30, 2)
    def set_ORD1_明細件数(self, value):
        encode_packed(self._data, 30, 2, True, value)

    # ORD1-明細行 OCCURS 10 TIMES (要素長 22, offset 32)

    # ORD1-商品コード
    def get_ORD1_商品コード(self, i0):
        return decode_alphanumeric(self._data, 32 + i0 * 22, 8)
    def set_ORD1_商品コード(self, i0, value):
        encode_alphanumeric(self._data, 32 + i0 * 22, 8, value)

    # ORD1-数量
    def get_ORD1_数量(self, i0):
        return decode_packed(self._data, 40 + i0 * 22, 3)
    def set_ORD1_数量(self, i0, value):
        encode_packed(self._data, 40 + i0 * 22, 3, True, value)

    # ORD1-単価
    def get_ORD1_単価(self, i0):
        return decode_packed(self._data, 43 + i0 * 22, 5)
    def set_ORD1_単価(self, i0, value):
        encode_packed(self._data, 43 + i0 * 22, 5, True, value)

    # ORD1-金額
    def get_ORD1_金額(self, i0):
        return decode_packed(self._data, 48 + i0 * 22, 6)
    def set_ORD1_金額(self, i0, value):
        encode_packed(self._data, 48 + i0 * 22, 6, True, value)

    # ORD1-処理区分
    def get_ORD1_処理区分(self):
        return decode_alphanumeric(self._data, 252, 1)
    def set_ORD1_処理区分(self, value):
        encode_alphanumeric(self._data, 252, 1, value)

    def is_ORD1_新規登録(self):
        return self.get_ORD1_処理区分() == "1"

    def is_ORD1_訂正(self):
        return self.get_ORD1_処理区分() == "2"

    def is_ORD1_取消(self):
        return self.get_ORD1_処理区分() == "9"

File: src/engine/transpile/src/test/resources/golden/SYK001/ORD1_受注レコード.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK001 のデータ部レコード ORD1-受注レコード を逐語対訳した自動生成コード。 */
public final class ORD1_受注レコード {

    /** 01 ORD1-受注レコード 総バイト長 253。 */
    public static final int LENGTH = 253;

    private final byte[] data;

    public ORD1_受注レコード() {
        this.data = new byte[253];
    }

    public ORD1_受注レコード(byte[] source) {
        this.data = Arrays.copyOf(source, 253);
    }

    public byte[] data() {
        return data;
    }

    // ORD1-受注番号
    public String get_ORD1_受注番号() {
        return CobolRuntime.decodeAlphanumeric(data, 0, 10);
    }
    public void set_ORD1_受注番号(String value) {
        CobolRuntime.encodeAlphanumeric(data, 0, 10, value);
    }

    // ORD1-受注日
    public long get_ORD1_受注日() {
        return CobolRuntime.decodeZoned(data, 10, 8);
    }
    public void set_ORD1_受注日(long value) {
        CobolRuntime.encodeZoned(data, 10, 8, false, value);
    }

    // ORD1-受注日-YMD REDEFINES ORD1-受注日 (同一オフセット 10)

    // ORD1-受注日-年
    public long get_ORD1_受注日_年() {
        return CobolRuntime.decodeZoned(data, 10, 4);
    }
    public void set_ORD1_受注日_年(long value) {
        CobolRuntime.encodeZoned(data, 10, 4, false, value);
    }

    // ORD1-受注日-月
    public long get_ORD1_受注日_月() {
        return CobolRuntime.decodeZoned(data, 14, 2);
    }
    public void set_ORD1_受注日_月(long value) {
        CobolRuntime.encodeZoned(data, 14, 2, false, value);
    }

    // ORD1-受注日-日
    public long get_ORD1_受注日_日() {
        return CobolRuntime.decodeZoned(data, 16, 2);
    }
    public void set_ORD1_受注日_日(long value) {
        CobolRuntime.encodeZoned(data, 16, 2, false, value);
    }

    // ORD1-得意先コード
    public String get_ORD1_得意先コード() {
        return CobolRuntime.decodeAlphanumeric(data, 18, 6);
    }
    public void set_ORD1_得意先コード(String value) {
        CobolRuntime.encodeAlphanumeric(data, 18, 6, value);
    }

    // ORD1-受注金額合計
    public long get_ORD1_受注金額合計() {
        return CobolRuntime.decodePacked(data, 24, 6);
    }
    public void set_ORD1_受注金額合計(long value) {
        CobolRuntime.encodePacked(data, 24, 6, true, value);
    }

    // ORD1-明細件数
    public long get_ORD1_明細件数() {
        return CobolRuntime.decodePacked(data, 30, 2);
    }
    public void set_ORD1_明細件数(long value) {
        CobolRuntime.encodePacked(data, 30, 2, true, value);
    }

    // ORD1-明細行 OCCURS 10 TIMES (要素長 22, offset 32)

    // ORD1-商品コード
    public String get_ORD1_商品コード(int i0) {
        return CobolRuntime.decodeAlphanumeric(data, 32 + i0 * 22, 8);
    }
    public void set_ORD1_商品コード(int i0, String value) {
        CobolRuntime.encodeAlphanumeric(data, 32 + i0 * 22, 8, value);
    }

    // ORD1-数量
    public long get_ORD1_数量(int i0) {
        return CobolRuntime.decodePacked(data, 40 + i0 * 22, 3);
    }
    public void set_ORD1_数量(int i0, long value) {
        CobolRuntime.encodePacked(data, 40 + i0 * 22, 3, true, value);
    }

    // ORD1-単価
    public long get_ORD1_単価(int i0) {
        return CobolRuntime.decodePacked(data, 43 + i0 * 22, 5);
    }
    public void set_ORD1_単価(int i0, long value) {
        CobolRuntime.encodePacked(data, 43 + i0 * 22, 5, true, value);
    }

    // ORD1-金額
    public long get_ORD1_金額(int i0) {
        return CobolRuntime.decodePacked(data, 48 + i0 * 22, 6);
    }
    public void set_ORD1_金額(int i0, long value) {
        CobolRuntime.encodePacked(data, 48 + i0 * 22, 6, true, value);
    }

    // ORD1-処理区分
    public String get_ORD1_処理区分() {
        return CobolRuntime.decodeAlphanumeric(data, 252, 1);
    }
    public void set_ORD1_処理区分(String value) {
        CobolRuntime.encodeAlphanumeric(data, 252, 1, value);
    }

    public boolean is_ORD1_新規登録() {
        return get_ORD1_処理区分().equals("1");
    }

    public boolean is_ORD1_訂正() {
        return get_ORD1_処理区分().equals("2");
    }

    public boolean is_ORD1_取消() {
        return get_ORD1_処理区分().equals("9");
    }
}
